Library/Homebrew/cmd/test-bot.rb

```diff
@@ -425,6 +425,10 @@ def formula formula_name
       install_args = %w[--verbose]
       install_args << "--build-bottle" unless ARGV.include? "--no-bottle"
       install_args << "--HEAD" if ARGV.include? "--HEAD"
+      # Pass --devel or --HEAD to the test-bot in the event formulae lack stable. Supports devel-only/head-only.
+      # head-only should not have devel, but devel-only can have head. Stable can have all three.
+      install_args << "--devel" if formula.devel && !formula.stable?
+      install_args << "--HEAD" if formula.head && !formula.stable? && !formula.devel?
```

Comment:
I can't quite wrap my head around how to move this into the below block whilst still preserving the "Don't duplicate the function" element introduced around line 478 with ` if formula.devel && formula.stable? && !ARGV.include?('--HEAD') \ `. I'm having a stupid hour, hopefully it'll come to me soon.

For now, I've pushed a change that makes it more readable, and less loads-of-instructions-packed-tightly-in-a-list-esque.